ias-verify/src/netscape_comment.rs

```diff
@@ -0,0 +1,53 @@
+use crate::utils::{length_from_raw_data, safe_indexing};
+use crate::CertDer;
+use frame_support::ensure;
+use std::convert::TryFrom;
+
+pub struct NetscapeComment<'a> {
+    pub attestation_raw: &'a [u8],
+    pub sig: Vec<u8>,
+    pub sig_cert: Vec<u8>,
+}
+
+pub const NS_CMT_OID: &[u8; 11] = &[
+    0x06, 0x09, 0x60, 0x86, 0x48, 0x01, 0x86, 0xF8, 0x42, 0x01, 0x0D,
+];
+
+impl<'a> TryFrom<CertDer<'a>> for NetscapeComment<'a> {
+    type Error = &'static str;
+
+    fn try_from(value: CertDer<'a>) -> Result<Self, Self::Error> {
+        // Search for Netscape Comment OID
+        let cert_der = value.0;
+
+        let mut offset = cert_der
+            .windows(NS_CMT_OID.len())
+            .position(|window| window == NS_CMT_OID)
+            .ok_or("Certificate does not contain 'ns_cmt_oid'")?;
+
+        offset += 12; // 11 + TAG (0x04)
+
+        #[cfg(test)]
+        println!("netscape");
+        // Obtain Netscape Comment length
+        let len = length_from_raw_data(cert_der, &mut offset)?;
+        // Obtain Netscape Comment
+        offset += 1;
+        let netscape_raw = safe_indexing(cert_der, offset, offset + len)?
+            .split(|x| *x == 0x7C)
+            .collect::<Vec<&[u8]>>();
+
+        ensure!(netscape_raw.len() == 3, "Invalid netscape payload");
```

Comment:
This is actually the fix for #35. Before, we iterated here three times and unwrapped without checking it `next()` yielded `Some`